src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -465,5 +481,41 @@ public virtual MethodDesc GetFinalizer()
         {
             return null;
         }
+
+        /// <summary>
+        /// Gets a value indicating whether this type has generic variance (the definition of the type
+        /// has a generic parameter that is co- or contravariant).
+        /// </summary>
+        public bool HasVariance
+        {
+            get
+            {
+                if ((_typeFlags & TypeFlags.HasGenericVarianceComputed) == 0)
+                {
+                    var flagsToAdd = TypeFlags.HasGenericVarianceComputed;
+
+                    foreach (GenericParameterDesc genericParam in GetTypeDefinition().Instantiation)
+                    {
+                        if (genericParam.Variance != GenericVariance.None)
+                        {
+                            flagsToAdd |= TypeFlags.HasGenericVariance;
+                            break;
+                        }
+                    }
+
```

Comment:
Could you pull this loop into a ComputeHasVariance function that is virtual? I believe for InstantiatedType it should return GetTypeDefinition().HasVariance, and the implementation on TypeDesc it should contain the code here.